crates/ln-dlc-node/src/ldk_node_wallet.rs

```diff
@@ -317,7 +280,10 @@ where
     }
 
     fn estimate_fee_rate(&self, confirmation_target: ConfirmationTarget) -> FeeRate {
-        let locked_fee_rate_cache = self.fee_rate_cache.read().unwrap();
+        let locked_fee_rate_cache = tokio::task::block_in_place(move || {
+            self.runtime_handle
+                .block_on(async move { self.fee_rate_cache.read().await })
+        });
```

Comment:
Played with this for a few minutes, could not easily get it working because of 

```
returns a reference to data owned by the current function
    |                                        temporary value created here
```

leaving this for now because it's not important.